test-plan/eventing/channel-lifecycle-conformance.md

```diff
@@ -0,0 +1,229 @@
+
+```
+kubectl get channel.messaging.knative.dev conformance-channel -o jsonpath='{.spec.channelTemplate.kind}'
+```
+
+Try to patch the annotation: `messaging.knative.dev/channel.spec.channelTemplate` to see if the resource mutates: 
+
+```
+kubectl patch channel.messaging.knative.dev conformance-channel --type merge -p '{"spec":{"channelTemplate":{"kind":"mutable"}}}'
+```
+
+You should get the following error: 
+```
+Error from server (BadRequest): admission webhook "validation.webhook.eventing.knative.dev" denied the request: validation failed: Immutable fields changed (-old +new): spec
+{v1.ChannelSpec}.ChannelTemplate.TypeMeta.Kind:
+	-: "InMemoryChannel" // or your channel implementation
+	+: "mutable"
+```
+
+Tested in eventing:
+- Not tested (depends on the implementation)
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/immutability"
+  "output": {
+    "channel Implementation": "<CHANNEL IMPLEMENTATION>",
+	  "expectedError": "<EXPECTED ERROR>"
+  }
+}
+```
+
+## [Test] Channel Readiness 
+
+Check for condition type `Ready` with status `True`: 
+
+```
+ kubectl get channel.messaging.knative.dev conformance-channel -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Tested in eventing:
+- https://github.com/knative/eventing/blob/release-0.26/test/rekt/features/channel/control_plane.go#L102
+- https://github.com/knative/eventing/blob/release-0.26/test/conformance/channel_status_test.go#L29
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/channel-readiness"
+  "output": {
+  	"channelImplementation": "<CHANNEL IMPLEMENTATION>",
+    "expectedType": "Ready",
+    "expectedStatus": "True"
+  }
+}
+```
+
+## [Test] Channel is Addressable
+
+Running the following command should return a URL
+
+```
+kubectl get channel.messaging.knative.dev conformance-channel -ojsonpath="{.status.address.url}"
+```
+
+Tested in eventing:
+- https://github.com/knative/eventing/blob/release-0.26/test/rekt/features/channel/control_plane.go#L102
+- https://github.com/knative/eventing/blob/release-0.26/test/conformance/channel_status_test.go#L29
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/channel-addressable"
+  "output": {
+  	"channelImplementation": "",
+	  "obtainedURL": "<CHANNEL URL>"
+  }
+}
+```
+
+## [Pre] Create Subscription for the Channel
+
+First lets create a Service that works as a Subscriber and a deadLetterSink for the Subscription:
+
+```
+kubectl apply -f control-plane/channel-lifecycle/services.yaml
+```
+
+Create a Subscription that points to the Channel:
+
+```
+kubectl apply -f control-plane/channel-lifecycle/subscription.yaml
+```
+
+## [Test] Channel Reference in Subscription
+
+Check that the Subscription is making a reference to the Channel, this should return the name of the Channel.
+
+```
+kubectl get subscription conformance-subscription -ojsonpath="{.spec.channel.name}"
+```
```

Comment:
We are reading what we just set when creating the subscription.
If the Spec says `Subscriptions MUST be associated with a Channel based on the spec.channel` I think it would be better to check the `AddedToChannel` status condition.

```
kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"AddedToChannel\")].status}"
```